miden-tx/src/host/account_delta_tracker.rs

```diff
@@ -89,38 +88,20 @@ impl AccountStorageDeltaTracker {
     /// Consumes `self` and returns the [AccountStorageDelta] that represents the changes to
     /// the account's storage.
     pub fn into_delta(self) -> AccountStorageDelta {
-        let mut cleared_items = Vec::new();
-        let mut updated_items = Vec::new();
-        let mut updated_maps: Vec<(u8, StorageMapDelta)> = Vec::new();
-
-        for (idx, value) in self.slot_updates {
-            if value == EMPTY_WORD {
-                cleared_items.push(idx);
-            } else {
-                updated_items.push((idx, value));
-            }
-        }
-
-        for (idx, map_deltas) in self.maps_updates {
-            let mut updated_leafs = Vec::new();
-            let mut cleared_leafs = Vec::new();
-
-            for map_delta in map_deltas {
-                if map_delta.1 == EMPTY_WORD {
-                    cleared_leafs.push(map_delta.0);
-                } else {
-                    updated_leafs.push(map_delta);
-                }
-            }
-            let storage_map_delta = StorageMapDelta::from(cleared_leafs, updated_leafs);
-            updated_maps.push((idx, storage_map_delta));
-        }
-
-        AccountStorageDelta {
-            cleared_items,
-            updated_items,
-            updated_maps,
-        }
+        let updated_maps = self
+            .maps_updates
+            .into_iter()
+            .map(|(idx, map_deltas)| {
+                (
+                    idx,
+                    StorageMapDelta::new(
+                        map_deltas.into_iter().map(|(key, value)| (key.into(), value)).collect(),
+                    ),
+                )
+            })
+            .collect();
```

Comment:
Should we define `AccountStorageDeltaTracker as:
```rust
pub struct AccountStorageDeltaTracker {
    slot_updates: BTreeMap<u8, Word>,
    maps_updates: BTreeMap<u8, StorageMapDelta>,
}
```
Then, this code would not be needed.